src/util/index.ts

```diff
@@ -1,2 +1,162 @@
+import './lang_string';
+import './misc';
+import './path';
+
+const {
+    sin,
+    getElementOffset,
+    removeFromArray,
+    toFixed,
+    transformPoint,
+    invertTransform,
+    getNodeCanvas,
+    createCanvasElement,
+    toDataURL,
+    multiplyTransformMatrices,
+    applyTransformToObject,
+    degreesToRadians,
+    enlivenObjects,
+    enlivenObjectEnlivables,
+    cleanUpJsdomNode,
+    loadImage,
+    setImageSmoothing,
+    getById,
+    addClass,
+    parsePreserveAspectRatioAttribute,
+    findScaleToFit,
+    findScaleToCover,
+    stylesFromArray,
+    stylesToArray,
+    hasStyleChanged,
+    getPathSegmentsInfo,
+    getPointOnPath,
+    string: {
+        graphemeSplit,
+        capitalize,
+        escapeXml
+    },
+    projectStrokeOnPoints,
+    array: {
+        min,
+        max
+    },
+    makePathSimpler,
+    parsePath,
+    joinPath,
+    getBoundsOfCurve,
+    limitDimsByArea,
+    capValue: clamp,
+    populateWithProperties,
+    qrDecompose,
+    saveObjectTransform,
+    resetObjectTransform,
+    object: {
+        clone,
+        extend
+    },
+    matrixToSVG,
+    sizeAfterTransform,
+    animate,
+    animateColor,
+    requestAnimFrame,
+    cancelAnimFrame,
+    rotateVector,
+    getRandomInt,
+    getSmoothPathFromPoints,
+    parseUnit,
+    toArray,
+    request,
+    addListener,
+    removeListener,
+    isTouchEvent,
+    sendPointToPlane,
+    radiansToDegrees,
+    setStyle,
+    calcRotateMatrix,
+    makeBoundingBoxFromPoints,
+    composeMatrix,
+    rotatePoint,
+} = fabric.util;
+export {
+    sin,
+    getElementOffset,
+    removeFromArray,
+    toFixed,
+    transformPoint,
+    invertTransform,
+    getNodeCanvas,
+    createCanvasElement,
+    toDataURL,
+    multiplyTransformMatrices,
+    applyTransformToObject,
+    degreesToRadians,
+    enlivenObjects,
+    enlivenObjectEnlivables,
+    cleanUpJsdomNode,
+    loadImage,
+    setImageSmoothing,
+    getById,
+    addClass,
+    parsePreserveAspectRatioAttribute,
+    findScaleToFit,
+    findScaleToCover,
+    stylesFromArray,
+    stylesToArray,
+    hasStyleChanged,
+    getPathSegmentsInfo,
+    getPointOnPath,
+    graphemeSplit,
+    capitalize,
+    escapeXml,
+    projectStrokeOnPoints,
+    min,
+    max,
+    makePathSimpler,
+    parsePath,
+    joinPath,
+    getBoundsOfCurve,
+    limitDimsByArea,
+    clamp,
+    populateWithProperties,
+    qrDecompose,
+    saveObjectTransform,
+    resetObjectTransform,
+    clone,
+    extend,
+    matrixToSVG,
+    sizeAfterTransform,
+    animate,
+    animateColor,
+    requestAnimFrame,
+    cancelAnimFrame,
+    rotateVector,
+    getRandomInt,
+    getSmoothPathFromPoints,
+    parseUnit,
+    toArray,
+    request,
+    addListener,
+    removeListener,
+    isTouchEvent,
+    sendPointToPlane,
+    radiansToDegrees,
+    setStyle,
+    calcRotateMatrix,
+    makeBoundingBoxFromPoints,
+    composeMatrix,
+    rotatePoint
+};
+export default fabric.util;
```

Comment:
let's not use default exports, is a mess when we mix default and named exports.
Let's merge this PR. But in general this kind of extraction i think is actual code more than a bunch of import/exports that goes away at rollup time. I would like to use named exports only unless we have some kind of concerns internally.

Reply:
For sure
I want named exports as well. 
Just for hacking the build

Reply:
well maybe we should remove this line
nothing should use it